official/utils/logs/logger.py

```diff
@@ -36,33 +37,96 @@
 _DATE_TIME_FORMAT_PATTERN = "%Y-%m-%dT%H:%M:%S.%fZ"
 
 
-class BenchmarkLogger(object):
-  """Class to log the benchmark information to local disk."""
+# Don't use it directly. Use get_benchmark_logger to access a logger.
+_benchmark_logger = None
+_logger_lock = threading.Lock()
 
-  def __init__(self, logging_dir):
-    self._logging_dir = logging_dir
-    if not tf.gfile.IsDirectory(self._logging_dir):
-      tf.gfile.MakeDirs(self._logging_dir)
 
-  def log_estimator_evaluation_result(self, eval_results):
-    """Log the evaluation result for a estimator.
+def config_benchmark_logger(logging_dir):
+  """Config the global benchmark logger"""
+  _logger_lock.acquire()
+  try:
+    global _benchmark_logger
```

Comment:
Why do we have to declare this as global?

Reply:
You have to put the global here to force python realize that you are trying to modify the instance above, rather than declare a local variable.

Reply:
Can you at least comment that here? Along with any side effects we should be aware of. (ie, what happens if I have a file with a local var _benchmark_logger?)

Reply:
The global scope is within this file/module, so this should be safe enough. If you want to access this field from other file, you usually just do logger. _benchmark_logger